src/redux/auth-reducer.tsx

```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
 const SET_USER_DATA = "SET_USER_DATA";
 const REGISTRATION_SUCCESS = "REGISTRATION_SUCCESS";
 
-export const setAuthUserData = (id, email, login, isAuth) => {
+type payloadType = {
+  id: null | number;
```

Comment:
usually it's better to use undefined for values that can be omitted and write `id: number | undefined;` that can be shorted to just `id: number?;`